source/elements/oneDNN/source/data_model/memory/formats.rst

```diff
@@ -0,0 +1,218 @@
+..
+  Copyright 2019-2020 Intel Corporation
+
+.. default-domain:: cpp
+
+.. cpp:namespace:: dnnl::memory
+
+.. |N| replace:: :math:`N`
+.. |C| replace:: :math:`C`
+.. |D| replace:: :math:`D`
+.. |H| replace:: :math:`H`
+.. |W| replace:: :math:`W`
+.. |G| replace:: :math:`G`
+.. |I| replace:: :math:`I`
+.. |O| replace:: :math:`O`
+
+.. |A| replace:: :math:`A`
+.. |M| replace:: :math:`M`
+.. |LDA| replace:: :math:`LDA`
+
+.. |n| replace:: :math:`n`
+.. |c| replace:: :math:`c`
+.. |d| replace:: :math:`d`
+.. |h| replace:: :math:`h`
+.. |w| replace:: :math:`w`
+
+.. |dimensions| replace:: :math:`\operatorname{dimensions}`
+.. |strides| replace:: :math:`\operatorname{strides}`
+.. |offset| replace:: :math:`\operatorname{offset}`
+
+.. _memory_formats-label:
+
+Memory Formats
+--------------
+
+In oneDNN memory format is how a multidimensional tensor is stored in
+1-dimensional linear memory address space. oneDNN specifies two kinds of
+memory formats: *plain* which correspond to traditional multidimensional
+arrays, and *optimized* which are completely opaque.
+
+Plain Memory Formats
+~~~~~~~~~~~~~~~~~~~~
+
+Plain memory formats describe how multidimensional tensors are laid out in
+memory using an array of |dimensions| and an array of |strides| both of which
+have length equal to the rank of the tensor. In oneDNN the order of dimensions
+is fixed and different dimensions can have certain canonical interpretation
+depending on the primitive. For example, for CNN primitives the order for
+activation tensors is :math:`\{N, C, ..., D, H, W\}`, where |N| stands for
+minibatch, |C| stands for channels, and |D|, |H|, and |W| stand for image
+spatial dimensions: depth, height and width respectively. Spatial dimensions
+may be omitted in the order from outermost to innermost; for example, it is
+not possible to omit |H| when |D| is present and it is never possible to omit
+|W|. Canonical interpretation is documented for each primitive. However, this
+means that the |strides| array plays an important role defining the order in
+which different dimension are laid out in memory. Moreover, the |strides| need
+to agree with |dimensions|.
+
+More precisely, let :math:`T` be a tensor of rank :math:`n` and let
+:math:`\sigma` be the permutation of the |strides| array that sorts it, i.e.
+:math:`\operatorname{strides}[\sigma(i)] \leq
+\operatorname{strides}[\sigma(j)]` if :math:`\sigma(i) < \sigma(j)` for all
+:math:`0 \leq i, j < n`. Then the following must hold:
+
+.. math::
+   \operatorname{stides}[j] \geq
+   \operatorname{strides}[j + 1] * \operatorname{dimensions}[\sigma(j)]
+   \text{ for all } 0 \leq j < n - 1.
```

Comment:
I am not sure I follow... I would expect `stride[j]` be always multiplied by `dim[j]`. However, the relation could invoke sigma. Something like:
```
Let sigma(i) be such that:
    sigma(i) > sigma(j) => stride[i] <= stride[j]
then
    stride[i] * dim[i] <= stride[j] for all i,j : sigma(i) > sigma(j)
```

Reply:
>  I would expect stride[j] be always multiplied by dim[j]. 

Just to make clear this statement clear(er): I would expect the same indices for stride and dim, while in the text strides use j, but dims use sigma(j)

Reply:
Yeah. You are right. The original text assumes that the strides are sorted and they are not. Not sure I came around to writing it this way :) I've fixed it but changed the notation a bit. Please take a look.

Reply:
I've fixed this but changed the notation a bit.

Reply:
LGTM, thanks